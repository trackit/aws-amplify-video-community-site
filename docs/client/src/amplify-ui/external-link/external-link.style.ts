import {css} from "emotion";

export const externalLinkStyle = css`
  display: inline;
  position: relative;
`;

export const graphicStyle = css`
  margin-left: 0.25rem;
  height: 0.5rem;
`;
