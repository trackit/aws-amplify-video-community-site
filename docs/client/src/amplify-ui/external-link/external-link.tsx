import {Component, Prop, h} from "@stencil/core";
import {externalLinkStyle, graphicStyle} from "./external-link.style";
import {track, trackExternalLink, AnalyticsEventType} from "../../utils/track";

@Component({tag: "amplify-external-link", shadow: false})
export class AmplifyExternalLink {
  /*** Whether to cancel the overriden target behavior (defaults to "_blank") */
  @Prop() readonly redirect?: boolean;
  /*** The URL to open / to which the anchor should redirect */
  @Prop() readonly href?: string;
  /*** Title attr same as for `a` */
  @Prop() readonly anchorTitle?: string;
  /*** Whether to display the external link graphic to the right of the anchor child text */
  @Prop() readonly graphic?: "black" | "white";

  onClick = () => {
    if (this.href) {
      track({
        type: AnalyticsEventType.EXTERNAL_LINK_CLICK,
        attributes: {from: location.href, to: this.href},
      });
      trackExternalLink(this.href);
    }
  };

  render() {
    return (
      <a
        class={externalLinkStyle}
        {...(this.redirect ? {} : {target: "_blank"})}
        onClick={this.onClick}
        href={this.href}
        title={this.anchorTitle}
        rel="noopener noreferrer"
      >
        <slot />
        {this.graphic && (
          <img
            class={graphicStyle}
            src={`/assets/external-link-${this.graphic}.svg`}
          />
        )}
      </a>
    );
  }
}
