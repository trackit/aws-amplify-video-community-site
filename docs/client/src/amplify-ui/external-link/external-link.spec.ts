import {AmplifyExternalLink} from "./external-link";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-external-link", () => {
  describe("Component logic", () => {
    let amplifyExternalLink: AmplifyExternalLink;
    beforeEach(() => (amplifyExternalLink = new AmplifyExternalLink()));

    it("should init `href` as `undefined`", () =>
      expect(amplifyExternalLink.href).toBeUndefined());

    it("should init `anchorTitle` as `undefined`", () =>
      expect(amplifyExternalLink.anchorTitle).toBeUndefined());

    it("should init `redirect` as `undefined`", () =>
      expect(amplifyExternalLink.redirect).toBeUndefined());

    it("should init `graphic` as `undefined`", () =>
      expect(amplifyExternalLink.graphic).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifyExternalLink],
            html: `<amplify-external-link />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
