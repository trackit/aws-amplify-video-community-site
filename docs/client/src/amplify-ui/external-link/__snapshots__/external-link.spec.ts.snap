// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-external-link Render logic should render 1`] = `
<amplify-external-link>
  <!---->
  <a class="css-q4acy3" rel="noopener noreferrer" target="_blank"></a>
</amplify-external-link>
`;
